lib/matplotlib/tests/test_polar.py

```diff
@@ -506,3 +506,23 @@ def test_polar_errorbar(order):
         ax.errorbar(theta, r, xerr=0.1, yerr=0.1, capsize=7, fmt="o", c="seagreen")
         ax.set_theta_zero_location("N")
         ax.set_theta_direction(-1)
+
+
+def test_radial_limits_behavior():
+    # r=0 is kept as limit if positive data and ticks are used
+    # negative ticks or data result in negative limits
+    fig = plt.figure()
+    ax = fig.add_subplot(projection='polar')
+    assert ax.get_ylim() == (0, 1)
+    # upper limit is expanded to include the ticks, but lower limit stays at 0
+    ax.set_rticks([1, 2, 3, 4])
+    assert ax.get_ylim() == (0, 4)
+    # upper limit is autoscaled to data, but lower limit limit stays 0
+    ax.plot([1, 2], [1, 2])
+    assert ax.get_ylim() == (0, 2)
+    # negative ticks also expand the negative limit
+    ax.set_rticks([-1, 0, 1, 2])
+    assert ax.get_ylim() == (-1, 2)
+    # negative data also autoscales to negative limits
+    ax.plot([1, 2], [-1, -2])
+    ax.get_ylim() == (-2.2, 2.2)
```

Comment:
```suggestion
    assert ax.get_ylim() == (-2, 2)
```
We see different numbers in the tests than running normally because by default the tests use the "classic" style, which [sets the margins to zero](https://github.com/matplotlib/matplotlib/blob/2d319fefa998b35738413b1256e65fadc3544819/lib/matplotlib/mpl-data/stylelib/classic.mplstyle#L220) whereas modern Matplotlib has a default margin of 0.05.